cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/transaction/coprocessor/CConfigurationCacheSupplier.java

```diff
@@ -20,37 +20,33 @@
 import com.google.common.base.Supplier;
 import org.apache.hadoop.conf.Configuration;
 
-import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Supplies instances of {@link CConfigurationCache} implementations.
  */
 public class CConfigurationCacheSupplier implements CacheSupplier<CConfigurationCache> {
-  private static final Object lock = new Object();
-  private static AtomicReference<CConfigurationCache> instance = new AtomicReference<>();
-  private static AtomicInteger refCount = new AtomicInteger(0);
-
-  private final ReferenceCountedSupplier<CConfigurationCache> referenceCountedSupplier;
+  private static volatile AtomicReference<ReferenceCountedSupplier<CConfigurationCache>> referenceCountedSupplier
```

Comment:
This can be changed to -
```
private static final ReferenceCountedSupplier<CConfigurationCache> = 
             new ReferenceCountedSupplier<>(CConfigurationCache.class.getSimpleName(), 
                                           new Supplier<CConfigurationCache>() {
                                              @Override
                                              public CConfigurationCache get() {
                                                 return new CConfigurationCache(hConf, sysConfTablePrefix);
                                             }
                                     });
```

This way we can ensure that initialization happens only once, and don't need the `AtomicReference`.